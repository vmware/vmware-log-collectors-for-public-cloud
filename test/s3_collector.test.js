/*
Copyright 2018 VMware, Inc.
SPDX-License-Identifier: MIT
*/

const { lintTestEnv } = require('../test/helper.test');
const { s3SampleData } = require('./sample_files/s3_testdata');
const { s3SampleFileContent } = require('./sample_files/s3_testdata');

const {
  S3HttpCollector,
} = require('../src/index');

describe('S3HttpCollector', () => {
  const tagRegexMap = new Map(Object.entries({
    tag1: new RegExp('c(k+)c', 'i'),
  }));
  const collector = new S3HttpCollector(lintTestEnv, tagRegexMap);

  describe('processLogsJson', () => {
    it('should process logs JSON', () => {
      const logsJson = s3SampleData();
      collector.processLogsJson(logsJson);
      expect(logsJson)
        .toMatchSnapshot();
    });
  });

  describe('processS3Data', () => {
    it('should process data JSON', () => {
      const logsJson = s3SampleFileContent();
      const data = collector.processS3Data('test', JSON.stringify(logsJson));
      expect(data)
        .toMatchSnapshot();
    });
  });
});

