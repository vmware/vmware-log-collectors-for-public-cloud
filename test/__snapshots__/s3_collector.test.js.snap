// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`S3HttpCollector processLogsJson should process logs JSON 1`] = `
Object {
  "Records": Array [
    Object {
      "awsRegion": "us-west-2",
      "bucket_arn": "arn:aws:s3:::sourcebucket",
      "bucket_name": "sourcebucket",
      "bucket_ownerIdentity_principalId": "A3NL1KOZZKExample",
      "configurationId": "testConfigRule",
      "eventName": "ObjectCreated:Put",
      "eventSource": "aws:s3",
      "eventTime": "1970-01-01T00:00:00.000Z",
      "eventVersion": "2.0",
      "ingest_timestamp": 1538769915817,
      "log_type": "aws_s3",
      "object_eTag": "d41d8cd98f00b204e9800998ecf8427e",
      "object_key": "HappyFace.jpg",
      "object_size": 1024,
      "object_versionId": "096fKKXTRTtl3on89fVO.nfljtsv6qko",
      "requestParameters_SourceIPAddress": "127.0.0.1",
      "responseElements_Xamzid2": "FMyUVURIY8/IgAtTv8xRjskZQpcIZ9KG4V5Wp6S7S/JRWeUWerMUE5JgHvANOjpD",
      "responseElements_Xamzrequestid": "C3D13FE58DE4C810",
      "s3SchemaVersion": "1.0",
      "text": Object {
        "bucket": Object {
          "arn": "arn:aws:s3:::sourcebucket",
          "name": "sourcebucket",
          "ownerIdentity": Object {
            "principalId": "A3NL1KOZZKExample",
          },
        },
        "configurationId": "testConfigRule",
        "object": Object {
          "eTag": "d41d8cd98f00b204e9800998ecf8427e",
          "key": "HappyFace.jpg",
          "size": 1024,
          "versionId": "096fKKXTRTtl3on89fVO.nfljtsv6qko",
        },
        "s3SchemaVersion": "1.0",
      },
      "userIdentityPrincipalId": "AIDAJDPLRKLG7UEXAMPLE",
    },
  ],
}
`;

exports[`S3HttpCollector processS3Data should process data JSON 1`] = `
Array [
  Object {
    "bucket_name": "test",
    "ingest_timestamp": 1538769915817,
    "log_type": "aws_s3data",
    "text": Object {
      "0_agent": "Debian APT-HTTP/1.3 (0.8.16~exp12ubuntu10.21)",
      "0_bytes": 0,
      "0_referrer": "-",
      "0_remote_ip": "93.180.71.3",
      "0_remote_user": "-",
      "0_request": "GET /downloads/product_1 HTTP/1.1",
      "0_response": 304,
      "0_time": "17/May/2015:08:05:32 +0000",
      "10_agent": "-",
      "10_bytes": 3316,
      "10_referrer": "-",
      "10_remote_ip": "217.168.17.5",
      "10_remote_user": "-",
      "10_request": "GET /downloads/product_2 HTTP/1.1",
      "10_response": 200,
      "10_time": "17/May/2015:08:05:12 +0000",
      "11_agent": "Debian APT-HTTP/1.3 (0.9.7.9)",
      "11_bytes": 0,
      "11_referrer": "-",
      "11_remote_ip": "188.138.60.101",
      "11_remote_user": "-",
      "11_request": "GET /downloads/product_2 HTTP/1.1",
      "11_response": 304,
      "11_time": "17/May/2015:08:05:49 +0000",
      "12_agent": "Debian APT-HTTP/1.3 (0.8.16~exp12ubuntu10.16)",
      "12_bytes": 0,
      "12_referrer": "-",
      "12_remote_ip": "80.91.33.133",
      "12_remote_user": "-",
      "12_request": "GET /downloads/product_1 HTTP/1.1",
      "12_response": 304,
      "12_time": "17/May/2015:08:05:14 +0000",
      "13_agent": "Debian APT-HTTP/1.3 (1.0.1ubuntu2)",
      "13_bytes": 318,
      "13_referrer": "-",
      "13_remote_ip": "46.4.66.76",
      "13_remote_user": "-",
      "13_request": "GET /downloads/product_1 HTTP/1.1",
      "13_response": 404,
      "13_time": "17/May/2015:08:05:45 +0000",
      "14_agent": "Debian APT-HTTP/1.3 (0.8.16~exp12ubuntu10.21)",
      "14_bytes": 324,
      "14_referrer": "-",
      "14_remote_ip": "93.180.71.3",
      "14_remote_user": "-",
      "14_request": "GET /downloads/product_1 HTTP/1.1",
      "14_response": 404,
      "14_time": "17/May/2015:08:05:26 +0000",
      "15_agent": "Debian APT-HTTP/1.3 (0.9.7.9)",
      "15_bytes": 0,
      "15_referrer": "-",
      "15_remote_ip": "91.234.194.89",
      "15_remote_user": "-",
      "15_request": "GET /downloads/product_2 HTTP/1.1",
      "15_response": 304,
      "15_time": "17/May/2015:08:05:22 +0000",
      "16_agent": "Debian APT-HTTP/1.3 (0.8.16~exp12ubuntu10.17)",
      "16_bytes": 0,
      "16_referrer": "-",
      "16_remote_ip": "80.91.33.133",
      "16_remote_user": "-",
      "16_request": "GET /downloads/product_1 HTTP/1.1",
      "16_response": 304,
      "16_time": "17/May/2015:08:05:07 +0000",
      "17_agent": "Go 1.1 package http",
      "17_bytes": 319,
      "17_referrer": "-",
      "17_remote_ip": "37.26.93.214",
      "17_remote_user": "-",
      "17_request": "GET /downloads/product_2 HTTP/1.1",
      "17_response": 404,
      "17_time": "17/May/2015:08:05:38 +0000",
      "18_agent": "Debian APT-HTTP/1.3 (0.9.7.9)",
      "18_bytes": 0,
      "18_referrer": "-",
      "18_remote_ip": "188.138.60.101",
      "18_remote_user": "-",
      "18_request": "GET /downloads/product_2 HTTP/1.1",
      "18_response": 304,
      "18_time": "17/May/2015:08:05:25 +0000",
      "1_agent": "Debian APT-HTTP/1.3 (0.8.16~exp12ubuntu10.21)",
      "1_bytes": 0,
      "1_referrer": "-",
      "1_remote_ip": "93.180.71.3",
      "1_remote_user": "-",
      "1_request": "GET /downloads/product_1 HTTP/1.1",
      "1_response": 304,
      "1_time": "17/May/2015:08:05:23 +0000",
      "2_agent": "Debian APT-HTTP/1.3 (0.8.16~exp12ubuntu10.17)",
      "2_bytes": 0,
      "2_referrer": "-",
      "2_remote_ip": "80.91.33.133",
      "2_remote_user": "-",
      "2_request": "GET /downloads/product_1 HTTP/1.1",
      "2_response": 304,
      "2_time": "17/May/2015:08:05:24 +0000",
      "3_agent": "Debian APT-HTTP/1.3 (0.8.10.3)",
      "3_bytes": 490,
      "3_referrer": "-",
      "3_remote_ip": "217.168.17.5",
      "3_remote_user": "-",
      "3_request": "GET /downloads/product_1 HTTP/1.1",
      "3_response": 200,
      "3_time": "17/May/2015:08:05:34 +0000",
      "4_agent": "Debian APT-HTTP/1.3 (0.8.10.3)",
      "4_bytes": 490,
      "4_referrer": "-",
      "4_remote_ip": "217.168.17.5",
      "4_remote_user": "-",
      "4_request": "GET /downloads/product_2 HTTP/1.1",
      "4_response": 200,
      "4_time": "17/May/2015:08:05:09 +0000",
      "5_agent": "Debian APT-HTTP/1.3 (0.8.16~exp12ubuntu10.21)",
      "5_bytes": 0,
      "5_referrer": "-",
      "5_remote_ip": "93.180.71.3",
      "5_remote_user": "-",
      "5_request": "GET /downloads/product_1 HTTP/1.1",
      "5_response": 304,
      "5_time": "17/May/2015:08:05:57 +0000",
      "6_agent": "Debian APT-HTTP/1.3 (0.8.10.3)",
      "6_bytes": 337,
      "6_referrer": "-",
      "6_remote_ip": "217.168.17.5",
      "6_remote_user": "-",
      "6_request": "GET /downloads/product_2 HTTP/1.1",
      "6_response": 404,
      "6_time": "17/May/2015:08:05:02 +0000",
      "7_agent": "Debian APT-HTTP/1.3 (0.8.10.3)",
      "7_bytes": 332,
      "7_referrer": "-",
      "7_remote_ip": "217.168.17.5",
      "7_remote_user": "-",
      "7_request": "GET /downloads/product_1 HTTP/1.1",
      "7_response": 404,
      "7_time": "17/May/2015:08:05:42 +0000",
      "8_agent": "Debian APT-HTTP/1.3 (0.8.16~exp12ubuntu10.17)",
      "8_bytes": 0,
      "8_referrer": "-",
      "8_remote_ip": "80.91.33.133",
      "8_remote_user": "-",
      "8_request": "GET /downloads/product_1 HTTP/1.1",
      "8_response": 304,
      "8_time": "17/May/2015:08:05:01 +0000",
      "9_agent": "Debian APT-HTTP/1.3 (0.8.16~exp12ubuntu10.21)",
      "9_bytes": 0,
      "9_referrer": "-",
      "9_remote_ip": "93.180.71.3",
      "9_remote_user": "-",
      "9_request": "GET /downloads/product_1 HTTP/1.1",
      "9_response": 304,
      "9_time": "17/May/2015:08:05:27 +0000",
    },
  },
]
`;
